File: CodingSamples/Java/Platform/SerializationTest/src/Visitor.java
package tourism;

import java.util.*;

public class Visitor implements java.io.Serializable {

	String id;
	int visitCount;
	Date lastVisit;
	//field declared with transient modifier is not included
	//in the serialization stream
	transient long ticket;
	static final long serialVersionUID = 1L;

	public Visitor(String name) {
		id = name;
	}

	public final String getId() { return id; }

	public final int getVisitCount() { return visitCount; }

	public final Date getLastVisit() { return lastVisit; }

	public final long getTicket() { return ticket; }

	public void visit() {
		visitCount += 1;
		lastVisit = new Date();
		ticket = System.currentTimeMillis() % 1000000;
	}
}

File: CodingSamples/DotNet/Platform/SerializationTest/Tourism2/Visitor.cs
namespace Tourism;

public class Visitor
{
    public string Id { get;}

    public int VisitCount { get; set; }

    public DateTime LastVisit { get; set; }

    [System.Text.Json.Serialization.JsonIgnore]
    public long Ticket { get; set; }

    public Visitor(string id)
    {
        Id = id;
    }

    public void Visit()
    {
        VisitCount += 1;
        LastVisit = DateTime.Now;
        Ticket = LastVisit.Ticks % 1000000;
    }
}